internal/ingress/controller/nginx.go

```diff
@@ -754,23 +754,31 @@ func (n *NGINXController) setupSSLProxy() {
 
 // IsDynamicConfigurationEnough decides if the new configuration changes can be dynamically applied without reloading
 func (n *NGINXController) IsDynamicConfigurationEnough(pcfg *ingress.Configuration) bool {
-	var copyOfRunningConfig ingress.Configuration = *n.runningConfig
-	var copyOfPcfg ingress.Configuration = *pcfg
+	copyOfRunningConfig := &ingress.Configuration{}
+	copyOfPcfg := &ingress.Configuration{}
+
+	mergo.MergeWithOverwrite(copyOfRunningConfig, n.runningConfig)
+	mergo.MergeWithOverwrite(copyOfPcfg, pcfg)
 
 	copyOfRunningConfig.Backends = []*ingress.Backend{}
 	copyOfPcfg.Backends = []*ingress.Backend{}
 
-	return copyOfRunningConfig.Equal(&copyOfPcfg)
+	return copyOfRunningConfig.Equal(copyOfPcfg)
 }
 
-// ConfigureDynamically JSON encodes new Backends and POSTs it to an internal HTTP endpoint
+// configureDynamically JSON encodes new Backends and POSTs it to an internal HTTP endpoint
 // that is handled by Lua
-func (n *NGINXController) ConfigureDynamically(pcfg *ingress.Configuration) error {
```

Comment:
why are you changing this?

Reply:
to be able to test it without creating a new `NGINXController` instance. 
Also removes an export we should not expose.

Reply:
> to be able to test it without creating a new NGINXController instance.

Is this a big deal though? I'm already doing it at https://github.com/kubernetes/ingress-nginx/pull/2365/files#diff-b43320e0cdcbbb75b01313d88f637165L55 to test the other function.

> Also removes an export we should not expose.

👍 for not exporting this function

Reply:
> Is this a big deal though? I'

I just want to make sure we don't have targets in the endpoints. This is the biggest object in the json request

Reply:
I understand that, I was commenting on "to be able to test it without creating a new NGINXController instance." - it is a few lines of code to create a new `NGINXController` instance in test.